src/components/ContentNode.vue

```diff
@@ -273,10 +273,15 @@ function renderNode(createElement, references) {
       }, (
         renderListItems(node.items)
       ));
-    case BlockType.paragraph:
-      return createElement('p', {}, (
+    case BlockType.paragraph: {
```

Comment:
Here is what I meant by poking inside. I think its a pretty simple thing we can do that helps us in the long run later.

Reply:
It seems really specific and only needed because we need a certain style to be applied in a certain situation. Are you sure there isn't a cleaner way of achieving this with CSS only? Maybe something that `:has` or `:only-of-type` could help with?

Reply:
I could not think of a way to add optional spacing to the paragraph or the next element, when that paragraph has an img somewhere inside it.

The file structure we have to work with is:

Just an img

```html
<maybe-something />

<p>
  <picture>
     <img />

<something-else />
```

With a figure

```html
<maybe-something />

<p>
  <figure>
    <picture>
       <img />

<something-else />
```

when we have an error in the img

```html
<maybe-something />

<p>
   <img />

```

WIth that in mind, `:has` could work, though I was not sure about browser support at the time - check https://caniuse.com/?search=has, its very new still.

I tried the selector below and it works, though its too broad and I need to clean it up a bit:

```css
p:has(img) + *, * + p:has(img)
```

Reply:
Because `has` doesn't have support for Firefox yet, I think the safest option is modifying `BlockType.paragraph` as you did, @dobromir-hristov